controllers/util/k8sutil.go

```diff
@@ -65,21 +65,27 @@ const (
 	ExitCodeThatShouldRestartTheProcess = 42
 
 	//ForbidMirroringLabel is used to forbid mirroring for ceph resources such as CephBlockPool
-	ForbidMirroringLabel                   = "ocs.openshift.io/forbid-mirroring"
-	BlockPoolMirroringTargetIDAnnotation   = "ocs.openshift.io/mirroring-target-id"
-	RequestMaintenanceModeAnnotation       = "ocs.openshift.io/request-maintenance-mode"
-	CephRBDMirrorName                      = "cephrbdmirror"
-	OcsClientTimeout                       = 10 * time.Second
-	StorageClientMappingConfigName         = "storage-client-mapping"
-	ExternalClassLabelKey                  = "storageclass.ocs.openshift.io/is-external"
-	StorageConsumerMirroringInfoAnnotation = "ocs.openshift.io/consumer-mirroring-info"
-	ForceDeletionAnnotationKey             = "ocs.openshift.io/force-deletion"
-	RookForceDeletionAnnotationKey         = "rook.io/force-deletion"
-	BackwardCompatabilityInfoAnnotationKey = "ocs.openshift.io/backward-compatability-info"
+	ForbidMirroringLabel                                  = "ocs.openshift.io/forbid-mirroring"
+	BlockPoolMirroringTargetIDAnnotation                  = "ocs.openshift.io/mirroring-target-id"
+	RequestMaintenanceModeAnnotation                      = "ocs.openshift.io/request-maintenance-mode"
+	CephRBDMirrorName                                     = "cephrbdmirror"
+	OcsClientTimeout                                      = 10 * time.Second
+	StorageClientMappingConfigName                        = "storage-client-mapping"
+	ExternalClassLabelKey                                 = "storageclass.ocs.openshift.io/is-external"
+	StorageConsumerMirroringInfoAnnotation                = "ocs.openshift.io/consumer-mirroring-info"
+	ForceDeletionAnnotationKey                            = "ocs.openshift.io/force-deletion"
+	RookForceDeletionAnnotationKey                        = "rook.io/force-deletion"
+	BackwardCompatabilityInfoAnnotationKey                = "ocs.openshift.io/backward-compatability-info"
+	InternalDeploymentMode                 DeploymentType = "Internal"
+	ExternalDeploymentMode                 DeploymentType = "External"
+	ProviderDeploymentMode                 DeploymentType = "Provider"
```

Comment:
This should go into its own const section. This is the convention that is used for enum like string in golang